Sources/Injection.swift

```diff
@@ -148,13 +149,19 @@ public enum DependenciesBuilder {
 }
 
 /// A new instance of `T` is created each time it is injected. The internal container holds no reference to it.
-public func factory<T>(constructor: @escaping () -> T) -> [Dependency] {
+public func factory<T>(constructor: @escaping DependencyFactory<T>) -> [Dependency] {
     [DependencyContainer(isSingleton: false, factory: constructor)]
 }
 
-/// The same instance of `T` is returned each time it is injected. An instance like this has its lifetime connected to the internal container instance, which essencially is the app lifetime.
-public func singleton<T>(constructor: @escaping () -> T) -> [Dependency] {
-    [DependencyContainer(isSingleton: true, factory: constructor)]
+/// The same instance of `T` is returned each time it is injected. An instance like this has its lifetime connected to the internal container instance, which essencially is the app lifetime. Since this is gonna be used throughout the app lifetime, it is instantiated immediatly. This allows catching errors early on. There is a lazy version, checkout `lazySingleton`.
```

Comment:
```suggestion
/// The same instance of `T` is returned each time it is injected. An instance like this has its lifetime connected to the internal container instance, which essencially is the app lifetime. Since this is gonna be used throughout the app lifetime, it is instantiated immediately. This allows catching errors early on. 
/// - SeeAlso: for the lazy equivalent of this function, see `lazySingleton`.
```